src/ng/compile.js

```diff
@@ -1967,13 +1967,16 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
           for (i in elementControllers) {
             controller = elementControllers[i];
             var controllerResult = controller();
-            if (controllerResult !== controller.instance &&
-                controller === controllerForBindings) {
-              // Remove and re-install bindToController bindings
-              thisLinkFn.$$destroyBindings();
-              thisLinkFn.$$destroyBindings =
+            if (controllerResult !== controller.instance && (isObject(controllerResult) || isFunction(controllerResult))) {
```

Comment:
I'm not sure what you are asking for here. We should mimic the behavior of native javascript: If they return an object or function use that as the result of the instantiation, otherwise return "as normal":

```javascript
function returnsStr() {
  this.foo = "bar";
  return "baz";
}

new returnsStr() === {foo:"bar"} // "baz" return value is ignored
```